src/ContentAwareGenerator.php

```diff
@@ -70,6 +70,8 @@ public function generate($name, $parameters = [], $absolute = UrlGeneratorInterf
     {
         if ($name instanceof SymfonyRoute) {
             $route = $this->getBestLocaleRoute($name, $parameters);
+        } elseif (array_key_exists(RouteObjectInterface::ROUTE_OBJECT, $parameters) && $parameters[RouteObjectInterface::ROUTE_OBJECT] instanceof SymfonyRoute) {
```

Comment:
i think for strict BC, we need here (and in getRouteByContent) to check `RouteObjectInterface::OBJECT_BASED_ROUTE_NAME === $name`. otherwise if someone happened to have the route object in the parameters, they see an unexpected change.

Reply:
Fixed